include/solvers/heat_transfer.h

```diff
@@ -506,74 +596,191 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
+  /**
+   * @brief Approximate an integral by evaluating the integrand at specific
+   * points and summing the point values with specific weights
+   */
+  std::shared_ptr<Quadrature<dim>> cell_quadrature;
+  /**
+   * @brief Approximate an integral by evaluating the integrand at specific
+   * points and summing the point values with specific weights
+   */
   std::shared_ptr<Quadrature<dim - 1>> face_quadrature;
 
 
   // Solution storage:
+
+  /**
+   * @brief Store a subset of the DoF indices that are stored on a
+   * particular processor on a distributed architecture.
+   */
   IndexSet locally_owned_dofs;
+  /**
+   * @brief Store a subset of the DoF indices that contains both the
+   * locally_owned_dofs and the DoF indices on all ghost cells.
+   */
   IndexSet locally_relevant_dofs;
 
-  GlobalVectorType               evaluation_point;
-  GlobalVectorType               local_evaluation_point;
-  GlobalVectorType               newton_update;
-  GlobalVectorType               present_solution;
-  GlobalVectorType               system_rhs;
-  AffineConstraints<double>      nonzero_constraints;
-  AffineConstraints<double>      zero_constraints;
+  /**
+   * @brief The point at which the evaluation is performed.
+   */
+  GlobalVectorType evaluation_point;
+  /**
+   * @brief The local evaluation point. Ghosts cells are not considered in
+   * this evaluation.
+   */
+  GlobalVectorType local_evaluation_point;
+  /**
+   * @brief The direction \f$ \delta u^n \f$ used to perform the newton iteration
+   * in \f$ u^{n+1} = u^n + \alpha ^n \delta u ^n \f$. Where \f$ \alpha ^n \f$
+   * is the size of the step.
+   */
+  GlobalVectorType newton_update;
+  /**
+   * @brief A vector containing all the values of the solution.
+   */
+  GlobalVectorType present_solution;
+  /**
+   * @brief The right hand side vector.
+   */
+  GlobalVectorType system_rhs;
+  /**
+   * @brief Store the nonzero constraints that arise from several sources such
+   * as boundary conditions and hanging nodes in the mesh. See the deal.II
+   * documentation on constraints on degrees of freedom for more information.
+   */
+  AffineConstraints<double> nonzero_constraints;
+  AffineConstraints<double> zero_constraints;
+  /**
+   * @brief The system matrix.
+   */
   TrilinosWrappers::SparseMatrix system_matrix;
 
 
-  // Previous solutions vectors
+  /**
+   * @brief Previous solution vector.
+   */
   std::vector<GlobalVectorType> previous_solutions;
 
-  // Solution transfer classes
+  /**
+   * @brief A parallel::distributed::SolutionTransfer object implements the
+   * transfer of a discrete FE function (e.g. a solution vector) from one mesh
+   * to another that is obtained from the first by a single refinement and/or
+   * coarsening step. During interpolation the vector is reinitialized to the
+   * new size and filled with the interpolated values. This Deal.ii class is
+   * used for mesh_refinement and simulation restarts.
```

Comment:
I don't think this needs this much detail here, but if you feel like it does, why not